apps/ledger-live-desktop/src/renderer/families/evm/StakeFlowModal_deprecated/EthStakingModalBody.tsx

```diff
@@ -87,13 +87,13 @@ export function EthStakingModalBody({
 
   const checkBoxOnChange = useCallback(() => {
     const value = !doNotShowAgain;
-    global.localStorage.setItem(`${LOCAL_STORAGE_KEY_PREFIX}${account?.currency}`, `${value}`);
+    global.localStorage.setItem(`${LOCAL_STORAGE_KEY_PREFIX}${account?.id}`, `${value}`);
```

Comment:
I don't understand this line. In [~/renderer/modals/Receive/steps/StepReceiveFunds.tsx:246](https://github.com/LedgerHQ/ledger-live/blob/7ae063f393625c9a3838baa789520dd113443f7a/apps/ledger-live-desktop/src/renderer/modals/Receive/steps/StepReceiveFunds.tsx#L246) `${LOCAL_STORAGE_KEY_PREFIX}${receivedCurrencyId}` is being checked which is `account.currency.id` or `undefined`.
Should this be ?
```suggestion
    global.localStorage.setItem(`${LOCAL_STORAGE_KEY_PREFIX}${account?.currency.id}`, `${value}`);
```

Reply:
I have added a helper fn here that should derive the correct account id : https://github.com/LedgerHQ/ledger-live/pull/9570/files#diff-49ada43cb9827cd9b76c8f9f2048c4d66564aa22a47803ba6f40ef9f5a16d131
if there are exceptions, we can add them there and it can update everywhere. wdyt?

Reply:
so this file `StakeFlowModal_deprecated` is not being used anymore, originally it was `account?.id` so i just reverted it to it, i made the change to currency id, it doesn't matter much to be honest since the component is deprecated and that value is just to have a unique key on the local storage

Reply:
Ok, so just to be clear this `setItem` and the `getItem` in `~/renderer/modals/Receive/steps/StepReceiveFunds.tsx:246` are not related ?

Reply:
You had a typecheck issue because you need to test what type of account it is to access the `currency` field